common/source/docs/common-fettec-onewire.rst

```diff
@@ -0,0 +1,101 @@
+==============
+FETtec OneWire
+==============
+
+This is a half duplex serial protocol with a 2Mbit/s Baudrate created by Felix Niessen (former Flyduino KISS developer) from `FETtec <https://fettec.net/>`_.
+You can change many more settings of the ESC over OneWire than over DShot.
+It also has a better cyclic redundancy check (CRC) than Dshot offers.
+CRC checks if there are any signal errors that occur in the communication between autopilot and ESC; and between ESC and autopilot.
+It supports ESC telemetry, so information from the ESC status can be sent back to the autopilot:
+
+- Motor rotations per minute (RPM)
+- Input voltage (V)
+- Current draw (A)
+- Power consumption (W)
+- Temperature (°C)
+
+This information allows the autopilot to:
+
+- dynamicaly change the center frequency of the notch filters used to reduce frame vibration noise in the gyros
+- log the status of each ESC to the SDCard or internal Flash, for post flight analysis
+- send the status of each ESC to the Ground Station or companion computer for real-time monitoring
+
+Connecting
+==========
+
+The protocol supports up-to 25 ESCs, but the ArduPilot implementation of it limits this to 12.
+You can connect all ESCs to a single serial port using only a single signal wire from the autopilot's UART TX pin to the ESC's TLM pin and one GND wire.
+If you are using 4in1 ESCs you can connect up-to 3 of them in this fashion.
+
+TODO: add a figure here.
+
+FullDuplex with Diode
+=====================
+If halfduplex is not available or working there is a workaround with a diode using the normal full duplex.
+This is also working with other "non-halfduplex" Controller like ESP32.
+
+The TLM-Pin (ESC) and RX (FC) connects to the anode (black side) of the diode.
+TX (FC) connects to cathode (white striped side) of the diode.
+
+ (TX --|<-- RX, TLM)
+
+Connection and Halfduplex with STM
+==================================
+
+To have reliable 2Mbit/s Baudrate the GPIO should be set to PushPull.
+On STM it is a special mode that is initialized like:
+
+gpioInit.mode = LL_GPIO_MODE_ALTERNATE;
+gpioInit.Pull = LL_GPIO_MODE_ALTERNATE;
+gpio.Pull = LL_GPIO_PULL_UP;
+gpioInit.OutputType = LL_GPIO_OUTPUT_PUSHPULL;
+gpioInit.Alternate = LL_GPIO_AF_1;
+
+
+Configuring
+===========
+
+These parameters have to be changed (in MissionPlanner under CONFIG -> Full Parameter List).
+To configure it you need to set the :ref:`SERIALx_PROTOCOL <common-serial-options#serialx-protocol>` parameter to `FETtec_OneWire` (38) and :ref:`SERIALx_OPTIONS <common-serial-options#serialx-options-parameter>` parameter to `HalfDuplex` (4), where x is the number of the serial port you are using.
+
+For example, a serial 2 connection would need:
+
+```
+SERIAL2_PROTOCOL = 38
+SERIAL2_OPTIONS 160 # full duplex - Use with diode
+SERIAL2_OPTIONS 164 # half duplex - Use without diode -> TX from FC to TLM of FETtec ESC
+```
+
+Change SERVO*_FUNCTION to 33-38 according to the needed motor order e.g.:
+```
+MOT_PWM_MAX 2000
```

Comment:
these are not linked...not sure what format they are in to get this highlighting